lib/connectors/registry.rb

```diff
@@ -4,49 +4,53 @@
 # you may not use this file except in compliance with the Elastic License.
 #
 
+require 'active_support/core_ext/class/subclasses'
+require 'connectors'
+
 module Connectors
-  class Factory
-    attr_reader :connectors
+  class Registry
+    class << self
+      def registered?(name)
+        connectors.has_key?(name)
+      end
 
-    def initialize
-      @connectors = {}
-    end
+      def connector_class(name)
+        connectors[name]
+      end
 
-    def register(name, klass)
-      @connectors[name] = klass
-    end
+      def connector(name, configuration)
+        klass = connector_class(name)
+        if klass.present?
+          return klass.new(configuration: configuration)
+        end
+        raise "Connector #{name} is not yet registered. You need to register it before use"
+      end
 
-    def registered?(name)
-      @connectors.has_key?(name)
-    end
+      def registered_connectors
+        connectors.keys.sort
+      end
 
-    def connector_class(name)
-      @connectors[name]
-    end
+      private
 
-    def connector(name, configuration)
-      klass = connector_class(name)
-      if klass.present?
-        return klass.new(configuration: configuration)
+      def connectors
+        @connectors ||= load_connectors
       end
-      raise "Connector #{name} is not yet registered. You need to register it before use"
-    end
 
-    def registered_connectors
-      @connectors.keys.sort
+      def load_connectors
+        connectors = {}
+        connectors_in_yaml = App::Config.connectors.dup || []
```

Comment:
Particularly this is the worrying place for me - reference to `App::` namespace, plus `Utility::Logger` - 3 objects in codebase reference each other in a circle